website/server/libs/user/index.js

```diff
@@ -19,6 +20,12 @@ export async function get (req, res, { isV3 = false }) {
     userToJSON = user.toJSON();
   }
 
+  if (!user.isAdmin()) {
```

Comment:
hmm, won't the user returned from routes different than GET /user include the secret? Maybe we should add it to `private` [here](https://github.com/HabitRPG/habitica/blob/develop/website/server/models/user/hooks.js#L24) and only return it for the hero routes?

Reply:
I just tried using the `private` array but with that I wouldn't be able to read the User for `/hall/heroes/:heroId` or did I miss a flag for that?

Reply:
Sorry, yeah if you add it to the private field it won't show up by default but at least it'll apply to all methods where the user is returned. I'd add it to the array and maybe add a new method to return it that you can use where necessary? So the hall routes probably

Reply:
I'd need a hint on how / what method I could create to read the `secret` , is the `private` just filtered when it is serialized to json?

If so I'll try to add the method and use it to get the secret then